app/models/patient.rb

```diff
@@ -940,4 +944,156 @@ def new_submission_token
+  # These side effects are handled in the handle_update function
+  def monitoring_history_edit(history_data, diff_state)
+    patient_before = history_data[:patient_before]
+    history_data[:updates]&.keys&.each do |attribute|
+      updated_value = self[attribute]
+      next if patient_before[attribute] == updated_value
+
+      case attribute
+      when :monitoring
+        History.monitoring_status(history_data)
+
+        # If the record was in continuous exposure and then it was closed and continuous exposure was turned off
+        if !updated_value && patient_before[:continuous_exposure] && !continuous_exposure
+          History.monitoring_change(
+            patient: self,
+            created_by: 'Sara Alert System',
+            comment: 'System turned off Continuous Exposure because the record was moved to the closed line list.'
+          )
+        end
+      when :exposure_risk_assessment
+        History.exposure_risk_assessment(history_data)
+      when :monitoring_plan
+        History.monitoring_plan(history_data)
+      when :public_health_action
+        History.public_health_action(history_data)
+      when :assigned_user
+        History.assigned_user(history_data)
+      when :pause_notifications
+        History.pause_notifications(history_data)
+      when :last_date_of_exposure
+        History.last_date_of_exposure(history_data)
+      when :extended_isolation
+        History.extended_isolation(history_data)
+      when :continuous_exposure
+        History.continuous_exposure(history_data)
+      when :symptom_onset
+        History.symptom_onset(history_data)
+      when :case_status
+        History.case_status(history_data, diff_state)
+
+        # If Case Status was updated to one of the values meant for the Exposure workflow and the Public Health Action was reset.
+        if ['Suspect', 'Unknown', 'Not a Case'].include?(updated_value) && patient_before[:public_health_action] != 'None' && public_health_action == 'None'
+          message = monitoring ?
+          "System changed Latest Public Health Action from \"#{patient_before[:public_health_action]}\" to \"None\" so that the monitoree will appear on
+          the appropriate line list in the exposure workflow to continue monitoring." : "System changed Latest Public Health Action
+          from \"#{patient_before[:public_health_action]}\" to \"None\"."
+          History.monitoring_change(patient: self, created_by: 'Sara Alert System', comment: message)
+        end
+      when :jurisdiction_id
+        History.jurisdiction(history_data)
+      end
+    end
+
+    # NOTE: Changes in case_status may trigger changes in isolation, so add isolation history messages
+    # at the end, so that they will come after any case_status history messages
+    return unless !history_data[:updates][:isolation].nil? && self[:isolation] != patient_before[:isolation]
```

Comment:
`self[:isolation]` can just be `isolation` here.

Reply:
Addressed via no longer needing this line of code in 1408283.